python/python_direct/python_translate.cpp

```diff
@@ -1332,12 +1332,18 @@ class PythonTranslator : public OptInConstDispatch {
   // If input and output values share the same type, a LoadStoreOp will be
   // created instead of a CastOp.
   void handle(const LoadStoreOp* lsop) final {
-    // TODO short-circuit: lsop is a permutation.
+    // short-circuit: lsop is a permutation.
```

Comment:
@rdspring1 , a basic question for my understanding. What happens here if the output has both, a root domain and an an allocation domain? Or is that not a possibility?

Reply:
I briefly discussed it with @jjsjann123, who suggested adding an assert to check we don't have both here.

Reply:
With direct binding, if we are going to expose API to specify transformation on ID and update TV allocation domain directly, then it's inevitable that we could have lsop with output containing both.

I'm not saying we need to support that today in this function. But having an assert would avoid surprises down the road.